File: apps/backend/src/EnterpriseAgentOs.Domain/Interfaces/Channels/IChannelRepository.cs
namespace EnterpriseAgentOs.Domain.Interfaces.Channels;

public interface IChannelRepository
{
    // ---------- Channel Connections ----------
    Task<IReadOnlyList<EnterpriseAgentOs.Domain.Models.ChannelConnectionRecord>> ListConnectionsAsync(CancellationToken ct = default);
    Task<EnterpriseAgentOs.Domain.Models.ChannelConnectionRecord?> GetConnectionAsync(Guid id, CancellationToken ct = default);
    Task<EnterpriseAgentOs.Domain.Models.ChannelConnectionRecord> CreateConnectionAsync(EnterpriseAgentOs.Domain.Models.ChannelConnectionRecord record, CancellationToken ct = default);
    Task<EnterpriseAgentOs.Domain.Models.ChannelConnectionRecord?> UpdateConnectionAsync(Guid id, Action<EnterpriseAgentOs.Domain.Models.ChannelConnectionRecord> apply, CancellationToken ct = default);
    Task<bool> DeleteConnectionAsync(Guid id, CancellationToken ct = default);

    // ---------- Agent Channel Bindings ----------
    Task<IReadOnlyList<EnterpriseAgentOs.Domain.Models.AgentChannelBindingRecord>> ListBindingsAsync(Guid agentId, CancellationToken ct = default);
    Task<EnterpriseAgentOs.Domain.Models.AgentChannelBindingRecord?> GetBindingAsync(Guid bindingId, CancellationToken ct = default);
    Task<EnterpriseAgentOs.Domain.Models.AgentChannelBindingRecord> CreateBindingAsync(EnterpriseAgentOs.Domain.Models.AgentChannelBindingRecord record, CancellationToken ct = default);
    Task<EnterpriseAgentOs.Domain.Models.AgentChannelBindingRecord?> UpdateBindingAsync(Guid bindingId, Action<EnterpriseAgentOs.Domain.Models.AgentChannelBindingRecord> apply, CancellationToken ct = default);
    Task<bool> DeleteBindingAsync(Guid bindingId, CancellationToken ct = default);

    // ---------- Routing queries ----------
    Task<IReadOnlyList<EnterpriseAgentOs.Domain.Models.AgentChannelBindingRecord>> FindBindingsByConnectionAsync(Guid connectionId, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/IChannelRepository.java
package com.enterpriseagentos.backendjava.domain.features.agents;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public interface IChannelRepository {
    CompletableFuture<List<ChannelConnectionRecord>> listConnectionsAsync();
    CompletableFuture<ChannelConnectionRecord> getConnectionAsync(UUID id);
    CompletableFuture<ChannelConnectionRecord> createConnectionAsync(ChannelConnectionRecord record);
    CompletableFuture<ChannelConnectionRecord> updateConnectionAsync(UUID id, java.util.function.Consumer<ChannelConnectionRecord> apply);
    CompletableFuture<Boolean> deleteConnectionAsync(UUID id);
    CompletableFuture<List<AgentChannelBindingRecord>> listBindingsAsync(UUID agentId);
    CompletableFuture<AgentChannelBindingRecord> getBindingAsync(UUID bindingId);
    CompletableFuture<AgentChannelBindingRecord> createBindingAsync(AgentChannelBindingRecord record);
    CompletableFuture<AgentChannelBindingRecord> updateBindingAsync(UUID bindingId, java.util.function.Consumer<AgentChannelBindingRecord> apply);
    CompletableFuture<Boolean> deleteBindingAsync(UUID bindingId);
    CompletableFuture<List<AgentChannelBindingRecord>> findBindingsByConnectionAsync(UUID connectionId);
    CompletableFuture<List<ChannelConnectionRecord>> findConnectionsByChannelTypeAsync(String channelType);
}
